ext/opentelemetry-exporter-cloud-monitoring/src/opentelemetry/exporter/cloud_monitoring/__init__.py

```diff
@@ -0,0 +1,136 @@
+            record.metric.name
+        )
+        if descriptor_type in self._metric_descriptors:
+            return self._metric_descriptors[descriptor_type]
+        descriptor = {
+            "name": "display_name",
+            "type": descriptor_type,
+            "display_name": record.metric.name,
+            "description": record.metric.description,
+            "labels": [],
+        }
+        for key, value in record.labels:
+            if isinstance(value, str):
+                descriptor["labels"].append(
+                    LabelDescriptor(key=key, value_type="STRING")
+                )
+            elif isinstance(value, int):
+                descriptor["labels"].append(
+                    LabelDescriptor(key=key, value_type="INT64")
+                )
+            elif isinstance(value, bool):
+                descriptor["labels"].append(
+                    LabelDescriptor(key=key, value_type="BOOL")
+                )
+        if isinstance(record.aggregator, CounterAggregator):
+            descriptor["metric_kind"] = MetricDescriptor.MetricKind.GAUGE
+        else:
+            logger.warning(
+                "Unsupported aggregation type %s, ignoring it",
+                type(record.aggregator).__name__,
+            )
+            return None
+        if record.metric.value_type == int:
+            descriptor["value_type"] = MetricDescriptor.ValueType.INT64
+        elif record.metric.value_type == float:
+            descriptor["value_type"] = MetricDescriptor.ValueType.DOUBLE
+        descriptor = self.client.create_metric_descriptor(
+            self.project_name, MetricDescriptor(**descriptor)
+        )
+        self._metric_descriptors[descriptor_type] = descriptor
+        return descriptor
+
+    def export(
+        self, metric_records: Sequence[MetricRecord]
+    ) -> "MetricsExportResult":
+        all_series = []
+        for record in metric_records:
+            metric_descriptor = self._get_metric_descriptor(record)
+            if not metric_descriptor:
+                continue
+
+            series = TimeSeries()
+            self._add_resource_info(series)
+            series.metric.type = metric_descriptor.type
+            for key, value in record.labels:
+                series.metric.labels[key] = value
+
+            point = series.points.add()
+            if record.metric.value_type == int:
+                point.value.int64_value = record.aggregator.checkpoint
+            elif record.metric.value_type == float:
+                point.value.double_value = record.aggregator.checkpoint
+            seconds, nanos = divmod(
+                record.aggregator.last_update_timestamp, 1e9
+            )
+            point.interval.end_time.seconds = int(seconds)
+            point.interval.end_time.nanos = int(nanos)
+            all_series.append(series)
+        self._batch_write(all_series)
+        return MetricsExportResult.SUCCESS
```

Comment:
Is there a case where this won't return success?

Reply:
There shouldn't be, as long as the API calls don't crash. The other metric exporters have a similar pattern.